packages/flutter/test/cupertino/button_test.dart

```diff
@@ -885,6 +885,90 @@ void main() {
     expect(value, isTrue);
   });
+
+  testWidgets('CupertinoButton.mouseCursor behavior', (WidgetTester tester) async {
+    const SystemMouseCursor customCursor = SystemMouseCursors.grab;
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoButton(
+            onPressed: () {},
+            mouseCursor: customCursor,
+            child: const Text('Tap Me'),
+          ),
+        ),
+      ),
+    );
+
+    final TestGesture gesture = await tester.createGesture(
+      kind: PointerDeviceKind.mouse,
+      pointer: 1,
+    );
+    // Get the center of the CupertinoButton to ensure the pointer is over it
+    final Offset buttonCenter = tester.getCenter(find.text('Tap Me'));
+    await gesture.addPointer(location: buttonCenter);
+    await tester.pumpAndSettle();
+
+    // Expect the custom cursor to be active when the mouse is over the button
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), customCursor);
+  });
+
+  testWidgets('CupertinoButton.Filled.mouseCursor behavior', (WidgetTester tester) async {
+    const SystemMouseCursor customCursor = SystemMouseCursors.grab;
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoButton.filled(
+            onPressed: () {},
+            mouseCursor: customCursor,
+            child: const Text('Tap Me'),
+          ),
+        ),
+      ),
+    );
+
+    final TestGesture gesture = await tester.createGesture(
+      kind: PointerDeviceKind.mouse,
+      pointer: 1,
+    );
+    // Get the center of the CupertinoButton to ensure the pointer is over it
+    final Offset buttonCenter = tester.getCenter(find.text('Tap Me'));
+    await gesture.addPointer(location: buttonCenter);
+    await tester.pumpAndSettle();
+
+    // Expect the custom cursor to be active when the mouse is over the button
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), customCursor);
+  });
+
+  testWidgets('CupertinoButton.Tinted.mouseCursor behavior', (WidgetTester tester) async {
+    const SystemMouseCursor customCursor = SystemMouseCursors.grab;
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoButton.tinted(
+            onPressed: () {},
+            mouseCursor: customCursor,
+            child: const Text('Tap Me'),
+          ),
+        ),
+      ),
+    );
+
+    final TestGesture gesture = await tester.createGesture(
+      kind: PointerDeviceKind.mouse,
+      pointer: 1,
+    );
+    // Get the center of the CupertinoButton to ensure the pointer is over it
+    final Offset buttonCenter = tester.getCenter(find.text('Tap Me'));
+    await gesture.addPointer(location: buttonCenter);
+    await tester.pumpAndSettle();
+
+    // Expect the custom cursor to be active when the mouse is over the button
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), customCursor);
+  });
 }
```

Comment:
Can you follow https://github.com/flutter/flutter/pull/151788/files#diff-6c819d3ed61b4436b70dc3d4cbeac15e24bc3705d0c3acf7761e58bcece5cf29R509 and add a test for a `WidgetStateMouseCursor` that uses the disabled state? (again, no selected and focused states)

Reply:
Resolved in recent PR